### PR TITLE
Updates `WP_Theme_JSON_Resolver` and tests to be equal to core

### PR DESCRIPTION
**Pull request by oandregal**
Related https://github.com/WordPress/wordpress-develop/pull/1808

This PR adapts the Gutenberg code to be like WordPress core, so it's easier to identify changes. In [this commit](https://github.com/WordPress/wordpress-develop/commit/d9599addd19db2bdc714697da98f0205aea5d007), we made changes to the class in WordPress core that weren't ported to Gutenberg.

## How to test

Verify tests pass.

### CODE REVIEW

File: lib/compat/wordpress-5.9/translate-settings-using-i18n-schema.php [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+/**
+ * Translates the provided settings value using its i18n schema.
+ *
+ * @param string|string[]|array[]|object $i18n_schema I18n schema for the setting.
+ * @param string|string[]|array[]        $settings    Value for the settings.
+ * @param string                         $textdomain  Textdomain to use with translations.
+ *
+ * @return string|string[]|array[] Translated settings.
+ */
+function gutenberg_translate_settings_using_i18n_schema( $i18n_schema, $settings, $textdomain ) {
```
Comment (jorgefilipecosta) [inline lib/compat/wordpress-5.9/translate-settings-using-i18n-schema.php:17]: These functions are already part of the core. Are they part of 5.8 if yes we can just remove the Gutenberg versions and use core.

**Review by jorgefilipecosta — APPROVED** [review]
Works well on my tests and seems to match what is already in core.

File: lib/compat/wordpress-5.9/translate-settings-using-i18n-schema.php (hunk 1 above)
Comment (oandregal) [reply lib/compat/wordpress-5.9/translate-settings-using-i18n-schema.php:17]: As far as I saw, they were introduced in August, post-5.8 release.

File: lib/class-wp-theme-json-resolver-gutenberg.php [hunk 2]
```diff
@@ -191,50 +97,12 @@ private static function translate_theme_json_chunk( array $array_to_translate, $
...
-				$array_to_translate = _wp_array_get( $theme_json, $path, null );
-				if ( is_null( $array_to_translate ) ) {
-					continue;
-				}
-
-				$translated_array = self::translate_theme_json_chunk( $array_to_translate, $key, $context, $domain );
-				gutenberg_experimental_set( $theme_json, $path, $translated_array );
-			}
+		if ( null === self::$i18n_schema ) {
+			$i18n_schema       = wp_json_file_decode( __DIR__ . '/theme-i18n.json' );
```
Comment (Mamaduka) [inline lib/class-wp-theme-json-resolver-gutenberg.php:101]: I think this should be `gutenberg_json_file_decode`, or we should rename the function. It is causing a fatal error, and I'm not able to access the site editor.

File: lib/class-wp-theme-json-resolver-gutenberg.php [hunk 3]
```diff
@@ -191,50 +97,12 @@ private static function translate_theme_json_chunk( array $array_to_translate, $
...
+			self::$i18n_schema = null === $i18n_schema ? array() : $i18n_schema;
 		}
 
-		return $theme_json;
+		return translate_settings_using_i18n_schema( self::$i18n_schema, $theme_json, $domain );
```
Comment (Mamaduka) [inline lib/class-wp-theme-json-resolver-gutenberg.php:105]: This line as well.

```
PHP Fatal error:  Uncaught Error: Call to undefined function wp_json_file_decode() in gutenberg/wp-content/plugins/gutenberg/lib/class-wp-theme-json-resolver-gutenberg.php:101
```

**Pull request merged**